### PR TITLE
Update LICENSE and NOTICE files

### PR DESCRIPTION
**Pull request by snleee**
Followed Apache Spark's approach for handling LICENSE, NOTICE.
https://github.com/apache/spark/commit/f825847c82042a9eee7bd5cfab106310d279fc32

- Checked licenses for all libraries that we bundle
- Checked licenses for all js/css files that we bundle
- Removed dependencies of category X libraries (#3718)

### CODE REVIEW

**Comment by snleee** [comment]
After removing category X dependency and removing "pinot-perf" from distribution package, we are bundling `109` external jar files and `9` js/css libraries within source code. After this pr, anyone who's changing version of any library or adding a new library needs to update NOTICE & LICENSE files.

**Comment by snleee** [comment]
@felixcheung Can you check this when you find the time?

**Comment by Jackie-Jiang** [comment]
Awesome! You make my OCD so satisfied!

**Comment by snleee** [comment]
@kishoreg Another approach that we can take is to copy licenses into `LICENSE` file. http://www.apache.org/dev/licensing-howto.html#permissive-deps

Do you prefer that approach?

File: LICENSE [hunk 1]
```diff
@@ -199,4 +198,170 @@
...
    See the License for the specific language governing permissions and
-   limitations under the License.
\ No newline at end of file
+   limitations under the License.
+
+------------------------------------------------------------------------------------
+This project bundles some components that are also licensed under the Apache
+License Version 2.0:
+
+com.101tec:zkclient
```
[the remark this answers is not in the document]
Comment (snleee) [reply LICENSE]: I just checked Spark's source code and binary distribution. For the source code, they include `LINCENSE, NOTICE` file for source code distribution while they include `LICENCE-binary, NOTICE-binary` for their binary distribution. I will keep 2 different license files one for source code and the other for binary.

File: NOTICE [hunk 2]
```diff
@@ -1,37 +1,445 @@
...
 
-This product includes software developed at
-eclipse (https://eclipse.org/jetty)
-Licensed under the Apache License 2.0.
+Copyright (C) 2002 Kevin Atkinson (kevina@gnu.org). Verbatim copying
+and distribution of this entire article is permitted in any medium,
+provided this notice is preserved.
+--------------------------------------------------------------------------------
+
+Apache HttpCore
```
[the remark this answers is not in the document]
Comment (snleee) [reply NOTICE]: Separated out `LICENSE, NOTICE` into `LICENSE, NOTICE, LICENSE-binary, NOTICE-binary`. `LICENSE, NOTICE` will be included for source code release while `LICENSE-binary, NOTICE-binary` will be included for binary distribution.

**Comment by felixcheung** [comment]
My take is if it's MIT license it needs not be included:
http://www.apache.org/dev/licensing-howto.html
http://www.apache.org/dev/licensing-howto.html#permissive-deps
https://www.apache.org/legal/resolved.html#required-third-party-notices

**Comment by s-sahay** [comment]
> After removing category X dependency and removing "pinot-perf" from distribution package, we are bundling `109 external jar libraries` and `9 js/css libraries` within the source code. After this pr, anyone who's changing version of any library or adding a new library needs to update NOTICE & LICENSE files.
> 
> The extensive list is present at the following spreadsheet https://docs.google.com/spreadsheets/d/1lMEEvrA8SqHe0Yy0yC2Nd-nN5lg5sPaNwihrj_--y8Q/edit?usp=sharing

Great job on getting this done! It would be good to have the google doc linked either on the cwiki or the RTD(may be in the Developer Guide section).

**Comment by snleee** [comment]
> My take is if it's MIT license it needs not be included:
> http://www.apache.org/dev/licensing-howto.html
> http://www.apache.org/dev/licensing-howto.html#permissive-deps
> https://www.apache.org/legal/resolved.html#required-third-party-notices

@felixcheung @kishoreg 
```
Bundling a dependency which is issued under one of the following licenses is straightforward, assuming that said license applies uniformly to all files within the dependency:

BSD (without advertising clause)
MIT/X11
In LICENSE, add a pointer to the dependency's license within the distribution and a short note summarizing its licensing:

This product bundles SuperWidget 1.2.3, which is available under a
"3-clause BSD" license.  For details, see deps/superwidget/.
```

It seems that the pointer is needed for BSD, MIT licenses. The example above has the link pointer to the full license `deps/superwiget`. Am I reading this incorrectly? 

As long as we don't violate the license issue, I can go ahead for removing copied license files from `/licenses` for BSD, MIT.

**Comment by snleee** [comment]
@felixcheung The current pr includes the license with MIT and BSD under `/licenses`. If you don't have any other concerns, would you give me a shipit along with the final check?

File: LICENSE [hunk 3]
```diff
@@ -199,4 +198,29 @@
...
+pinot-controller/src/main/resources/*/js/lib/codemirror/*
+pinot-controller/src/main/resources/*/js/lib/foundation/*
+pinot-controller/src/main/resources/*/js/lib/angular*
+pinot-controller/src/main/resources/*/js/lib/underscore*
+pinot-controller/src/main/resources/*/js/lib/jquery-2.1.3.min.js
+pinot-controller/src/main/resources/*/js/lib/beautify.js
+pinot-controller/src/main/resources/*/js/lib/handlebars.js
+pinot-controller/src/main/resources/*/css/lib/codemirror*
+pinot-controller/src/main/resources/*/css/lib/foundation*
+pinot-controller/src/main/resources/*/css/lib/normalize.css
```
Comment (felixcheung) [inline LICENSE]: since I'm not too familiar with this - are we saying these js css files are checked into the git repo?

Comment (snleee) [reply LICENSE]: Those files are the part of the git repo (they are used for some UI components. e.g. pinot query console). After checking in this, I will file a separate PR for handling the build to include LICENSE-binary for the binary release.

**Pull request merged**